packages/liferay-js-insights/index.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const globby = require('globby');
+const findUp = require('find-up');
+const fs = require('fs');
+const util = require('util');
+const argv = require('minimist')(process.argv.slice(2));
+const parser = require('@babel/parser');
+const path = require('path');
+
+const readFile = util.promisify(fs.readFile);
+
+const INSIGHTS = [require('./insights/dependencies')];
+
+let report;
+
+if (argv.json) {
+	report = require('./reporters/json');
+} else if (argv.airtable) {
+	report = require('./reporters/airtable');
+} else {
+	report = require('./reporters/table');
+}
+
+/**
+ * Parses the contents of a file and returns the corresponding AST
+ */
+async function parse(path) {
+	const content = await readFile(path, 'utf8');
+
+	return parser.parse(content, {
+		plugins: ['jsx', 'classProperties'],
+		sourceType: 'module'
+	});
+}
+
+async function getModuleMeta(modulePath) {
+	// Resolves current modulePath base directory
+	const cwd = path.dirname(path.resolve(modulePath));
+
+	// Finds the closer package.json file to the modulePath
+	const pkg = await findUp('package.json', {cwd});
+
+	// Take app from the `name` field in package.json
+	const app = JSON.parse(await readFile(pkg, 'utf8')).name;
+
+	const name = modulePath.substring(modulePath.lastIndexOf('/') + 1);
+
+	// Finds the root git folder
+	const gitRoot = await findUp('.git', {
+		cwd,
+		type: 'directory'
+	});
```

Comment:
There is also `git rev-pa`se --show-toplevel` which handles some edge cases that you may or may not care about like:

- if you run in a submodule (unlikely to come up for us).
- if you run in a `git worktree` checkout  (probably relevant to us).

Reply:
I'll take a look! 👍